src/courtroom.cpp

```diff
@@ -1309,16 +1294,21 @@ void Courtroom::append_system_text(QString p_showname, QString p_line)
 
 void Courtroom::play_preanim()
 {
-  QString f_char = m_chatmessage[CMChrName];
-  QString f_preanim = m_chatmessage[CMPreAnim];
-
   // all time values in char.inis are multiplied by a constant(time_mod) to get
   // the actual time
-  int text_delay = ao_app->get_text_delay(f_char, f_preanim) * time_mod;
   int sfx_delay = m_chatmessage[CMSoundDelay].toInt() * 60;
-
   sfx_delay_timer->start(sfx_delay);
 
+  QString f_preanim = m_chatmessage[CMPreAnim];
+
+  if (f_preanim.trimmed() == "-")
+  {
+    // no animation, continue
+    preanim_done();
+    return;
+  }
```

Comment:
Can you elaborate a bit?

Reply:
As in, if we're going to have hardcoded keywords for it, we should prepare them way before.

When we first prepare and receive the message, this would reduce the need to check more data since we know most of them are taken into account for the message that we're trying to display.

The preparation phase should be either in `handle_chatmessage` or a method prior to it (`prepare_chatmessage`?)